tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,1020 @@
+       "        output_1: Tensor(torch.Size([5, 4]), dtype=torch.float32),\n",
+       "        output_2: Tensor(torch.Size([5, 10]), dtype=torch.float32)},\n",
+       "    batch_size=torch.Size([5]),\n",
+       "    device=cpu,\n",
+       "    is_shared=False)"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "tensordict = TensorDict({\"a\": torch.randn(5, 3)}, batch_size=[5])\n",
+    "\n",
+    "splitlinear = TensorDictModule(\n",
+    "    MultiHeadLinear(3, 4, 10),\n",
+    "    in_keys=[\"a\"],\n",
+    "    out_keys=[\"output_1\", \"output_2\"],\n",
+    ")\n",
+    "splitlinear(tensordict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "859630c3",
+   "metadata": {},
+   "source": [
+    "As we shown previously, the `TensorDictModule` can take any `nn.Module` and perform the operations on a `TensorDict`. When having multiple input keys and output keys, make sure they match the order in the module.\n",
+    "`TensorDictModule` can work with `TensorDict` instances that contain more tensors than what the `in_keys` attribute indicates. Unless a `vmap` operator is used, the `TensorDict` is modified in-place."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11d2d2a7-6a55-4f31-972b-041be387f9df",
+   "metadata": {},
+   "source": [
+    "### Example 4: Combining multiples `TensorDictModule` with `TensorDictSequence`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89b157d5-322c-45d6-bec9-20440b78a2bf",
+   "metadata": {},
+   "source": [
+    "To combine multiples `TensorDictModule`instances, we can une `TensorDictSequence`. This block will take the input of the n-1th `TensorDictModule` in a list and feed it to the nth `TensorDictModule`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "7e36d071-df67-4232-a8a9-78e79b32fef2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tensordict = TensorDict({\"a\": torch.randn(5, 3)}, batch_size=[5])\n",
+    "\n",
+    "splitlinear = TensorDictModule(\n",
+    "    MultiHeadLinear(3, 4, 10),\n",
+    "    in_keys=[\"a\"],\n",
+    "    out_keys=[\"output_1\", \"output_2\"],\n",
+    ")\n",
+    "mergelinear = TensorDictModule(\n",
+    "    MergeLinear(4, 10, 13), in_keys=[\"output_1\", \"output_2\"], out_keys=[\"output\"]\n",
+    ")\n",
+    "\n",
+    "split_and_merge_linear = TensorDictSequence(splitlinear, mergelinear)\n",
+    "\n",
+    "assert split_and_merge_linear(tensordict)['output'].shape == torch.Size([5, 13])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "760118ea",
+   "metadata": {},
+   "source": [
+    "### Example 5: Compatibility with functorch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e2718a12",
+   "metadata": {},
+   "source": [
+    "TensorDictModule is compatible with functorch. We can use make_functional_with_buffers on top of it."
```

Comment:
not really: I would rather say

"TensorDictModule comes with its own `make_functional_with_buffers` method"

What does "on top of it" mean in this context?